packages/synapse-react-client/src/components/SynapseTable/SynapseTable.tsx

```diff
@@ -1,889 +1,275 @@
+  /** If true and entity is a file view or dataset, renders a column that allows the logged-in user to directly download the file, if they have permission */
+  showDirectDownloadColumn?: boolean
+  /** If true, the add to download list column will be hidden */
+  hideAddToDownloadListColumn?: boolean
+  /** Configuration to override cell renderers with e.g. a link to a portals detail page */
   columnLinks?: LabelLinkConfig
-  hideDownload?: boolean
 }
 
-export class SynapseTable extends React.Component<
-  SynapseTableProps,
-  SynapseTableState
-> {
-  constructor(props: SynapseTableProps) {
-    super(props)
-    this.componentDidMount = this.componentDidMount.bind(this)
-    this.componentWillUnmount = this.componentWillUnmount.bind(this)
-    this.componentDidUpdate = this.componentDidUpdate.bind(this)
-    this.shouldComponentUpdate = this.shouldComponentUpdate.bind(this)
-    this.handleColumnSortPress = this.handleColumnSortPress.bind(this)
-    this.findSelectionIndex = this.findSelectionIndex.bind(this)
-    this.advancedSearch = this.advancedSearch.bind(this)
-    this.getLengthOfPropsData = this.getLengthOfPropsData.bind(this)
-    this.configureFacetDropdown = this.configureFacetDropdown.bind(this)
-    this.enableResize = this.enableResize.bind(this)
-    this.disableResize = this.disableResize.bind(this)
-    this.isEntityViewOrDataset = this.isEntityViewOrDataset.bind(this)
-    this.allRowsHaveId = this.allRowsHaveId.bind(this)
+const columnHelper = createColumnHelper<Row>()
 
-    // store the offset and sorted selection that is currently held
-    this.state = {
-      /* columnIconSortState tells what icon to display for a table
-         header. There are three states for a particular header-
-          0 - show descending icon but *deselected*
-          1 - show descending icon selected
-          2 - show ascending icon selected
-      */
-      columnIconSortState: [],
-      isExportTableDownloadOpen: false,
-      isExpanded: false,
-      isColumnSelectionOpen: false,
-      // sortedColumnSelection contains the columns which are
-      // selected currently and their sort status as eithet
-      // off, desc, or asc.
-      sortedColumnSelection: [],
-      mapEntityIdToHeader: {},
-      mapUserIdToHeader: {},
-      isFetchingEntityHeaders: false,
-    }
-    this.getEntityHeadersInData = this.getEntityHeadersInData.bind(this)
-  }
+export function SynapseTable(props: SynapseTableProps) {
+  const { showDownloadColumn } = props
+  const {
+    showAccessColumn,
+    showDirectDownloadColumn = showDownloadColumn,
+    hideAddToDownloadListColumn,
+    columnLinks,
+  } = props
+  const { getCurrentQueryRequest } = useQueryContext()
+  const { data, entity, isLoadingNewBundle, isRowSelectionVisible } =
+    useQueryContext()
+  const { columnsToShowInTable, NoContentPlaceholder } =
+    useQueryVisualizationContext()
+  const synapseTableContext = useMemo(() => ({ columnLinks }), [columnLinks])
```

Comment:
`columnLinks` needs to be passed to `SynapseTableCell` and there's no longer an easy way to do that, so just use a context object.